### DISCUSSION TITLE
Spice works on Ubuntu, not on Debian guests

### DISCUSSION THREAD

**Discussion by lelegard** [utmapp/UTM#5726] [Linux]
**Configuration**  
* UTM Version: 4.2.5
* macOS Version: 13.3.1
* Mac Chip: M1

Ubuntu 23.04 guest:
- Installed several months ago with a previous version of UTM.
- `spice-vdagent` 0.22.1-3 is installed.
- It works correctly:
  - The display is reconfigured on resize.
  - The shared folder is correctly mounted.

Debian 11.7 guest:
- Just installed using UTM 4.2.5.
- `spice-vdagent` 0.20.0-2 is installed.
- Does not work:
  - The display is not reconfigured on resize, it proportionally shrinks or enlarges.
  - The shared folder cannot be mounted

~~~
# modprobe 9pnet_virtio
# mount -t 9p -o trans=virtio share /shared
mount: /shared: special device share does not exist.
~~~

Is this due to Ubuntu vs Debian? Spice agent 0.22 vs. 0.20 (the standard versions on their respective distro)? Something else?

**Comment by dmarmor** [comment]
I'm having the same problem with the shared directory on Debian. I get the following error:

```bash
$ sudo mount -t 9p -o trans=virtio share test -oversion=9p2000.L
mount: /home/XX/XXXX: special device share does not exist.
       dmesg(1) may have more information after failed mount system call.
```

Here's the relevant output from dmesg:

```
9pnet_virtio: no channels available for device share
```

Any ideas how to fix this? Thanks!

**Comment by tosiara** [comment]
Same issue:
```
[    2.874179] 9pnet: Installing 9P2000 support
[    2.880451] 9p: Installing v9fs 9p2000 file system support
[    2.884672] 9pnet_virtio: no channels available for device share
```

**Comment by sc07kvm** [comment]
Same issue on guest Ubuntu 18.04.02(kernel 4.18.20):

```
[   39.213352] 9p: Installing v9fs 9p2000 file system support
[   39.213832] FS-Cache: Netfs '9p' registered for caching
[   39.215378] 9pnet_virtio: no channels available for device share
```

**Comment by ssuperuser** [comment]
I ran into similiar issue with using 22.04 ARM Ubuntu version, when I installed Ubuntu 23.04 ARM version directory started working without any issues. 

UTM Version 4.4.4 (92)
MACOS 14.0 (23A344)

root@ubuntu2304-arm-ds:~# mount -t virtiofs share /tmp/das
root@ubuntu2304-arm-ds:~# ls
snap
root@ubuntu2304-arm-ds:~# lsblk 
NAME                      MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
loop0                       7:0    0 68.5M  1 loop /snap/core22/867
                                                   /snap/core22/867
loop1                       7:1    0  169M  1 loop /snap/lxd/25855
                                                   /snap/lxd/25855
loop2                       7:2    0 35.5M  1 loop /snap/snapd/20298
                                                   /snap/snapd/20298
sda                         8:0    0  2.2G  1 disk 
├─sda1                      8:1    0  2.2G  1 part 
└─sda2                      8:2    0  6.6M  1 part 
vda                       253:0    0   64G  0 disk 
├─vda1                    253:1    0    1G  0 part /boot/efi
├─vda2                    253:2    0    2G  0 part /boot
└─vda3                    253:3    0 60.9G  0 part 
  └─ubuntu--vg-ubuntu--lv 252:0    0 30.5G  0 lvm  /snap
                                                   /
root@ubuntu2304-arm-ds:~# ls /tmp/das/
Documents
root@ubuntu2304-arm-ds:~# cd /tmp/das/Documents/
root@ubuntu2304-arm-ds:/tmp/das/Documents# ls
'~$tandup.docx'

**Comment by tosiara** [comment]
I found another workaround:
1. shutdown guest
2. go to UTM settings and change sharing folder to the same one
3. boot guest - works

Yes, it is just enough to open file manager and select the same shared dir and it resolves. You must do that after every MacOS reboot

**Reply by ArcadeHacker** [reply]
Same issue, same workaround. Mac x86 Ventura 13.6 (22G120) + UTM Version 4.3.5 (87) + Ubuntu 22.04.3 LTS (fully upgraded)

**Reply by drwhitehouse** [reply]
I can't get this workaround to work for me on Debian Trixie.
I assume the idea is having set things up and finding it doesn't work, you simply shut down the guest remove / replace the share and start it up. If that is not correct I've misunderstood this workaround but there still seem to be issues with Debian as far as I can tell.

**Comment by driftli** [comment]
Try to share files with virtiofs: https://libvirt.org/kbase/virtiofs.html
Don't forget the `<memoryBacking>` elements, and then reboot to take effect. I solved it by this.
(PS: My QEMU version is `6.2.0`, and my os is ubunut `22.04`. )
